### PR TITLE
add chart for aerospike

### PR DESCRIPTION
**Pull request by ghost**
This is an implementation of Aerospike StatefulSet found here:

 * https://github.com/aerospike/aerospike-kubernetes

### CODE REVIEW

File: stable/aerospike/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+description: A Helm chart for Aerospike in Kubernetes
+name: aerospike
+keywords:
+  - aerospike
+  - big-data
+home: http://aerospike.com
+version: 0.1.5
+icon: https://s3-us-west-1.amazonaws.com/aerospike-fd/wp-content/uploads/2016/06/Aerospike_square_logo.png
+sources:
+- https://github.com/aerospike/aerospike-server
```
Comment (unguiculus) [inline stable/aerospike/Chart.yaml:11]: Add `maintainers` and `appVersion`.

File: stable/aerospike/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# Aerospike Helm Chart
+
+This is an implementation of Aerospike StatefulSet found here:
+
+ * https://github.com/aerospike/aerospike-kubernetes
+
+## Pre Requisites:
+
+* Kubernetes 1.5 with beta APIs enabled and support for statefulsets
```
Comment (unguiculus) [inline stable/aerospike/README.md]: Kubernetes 1.7+ with beta APIs enabled

File: stable/aerospike/README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+To install the chart with the release name `my-aerospike` using a dedicated namespace(recommended):
+
+```
+$ helm repo add incubator http://storage.googleapis.com/kubernetes-charts-incubator
+$ helm install --name my-aerospike --namespace aerospike incubator/aerospike
+```
+
+The chart can be customized using the following configurable parameters:
+
```
Comment (unguiculus) [inline stable/aerospike/README.md:39]: Align table nicely.

File: stable/aerospike/templates/configmap.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "fullname" . }}
```
Comment (unguiculus) [inline stable/aerospike/templates/configmap.yaml]: Add standard labels.

File: stable/aerospike/templates/statefulset.yaml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    app: {{ template "name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  serviceName: {{ template "fullname" . }}-mesh
+  replicas: {{ .Values.replicaCount }}
+  template:
+    metadata:
+      labels:
```
Comment (unguiculus) [inline stable/aerospike/templates/statefulset.yaml]: Only `app` and `release` labels are usually used here.

File: stable/aerospike/templates/statefulset.yaml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+{{ toYaml .Values.affinity | indent 8 }}
+      {{- end }}
+      volumes:
+      - name: config-volume
+        configMap:
+          name: {{ template "fullname" . }}
+          items:
+          - key: aerospike.conf
+            path: aerospike.conf
+  volumeClaimTemplates:
```
Comment (unguiculus) [inline stable/aerospike/templates/statefulset.yaml]: We've started introducing a common pattern for persistence. See #1869. Pleae adapt and use for your PR. Here's a statefulset example:

https://github.com/kubernetes/charts/blob/master/stable/mongodb-replicaset/templates/mongodb-statefulset.yaml#L188-L214

File: stable/aerospike/values.yaml [hunk 7]
```diff
@@ -0,0 +1,99 @@
+# Default values for aerospike.
+terminationGracePeriodSeconds: 30
+replicaCount: 1
+nodeSelector: {}
+image:
+  repository: aerospike/aerospike-server
+  tag: 3.14.1.2
+  pullPolicy: Always
```
Comment (unguiculus) [inline stable/aerospike/values.yaml]: Use `IfNotPresent` as default.

File: stable/aerospike/templates/_helpers.tpl [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+*/}}
+{{- define "fullname" -}}
+{{- $name := default .Chart.Name .Values.nameOverride -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Create aerospike mesh setup
+*/}}
+{{- define "aerospike.mesh" -}}
```
Comment (unguiculus) [inline stable/aerospike/templates/_helpers.tpl:21]: We've started namespacing templates (see #1785). Please apply as well for `name` and `fullname`.

**Review by unguiculus — CHANGES_REQUESTED** [review]
Thanks for the PR and sorry for the delay. Please update with current master.

**Comment by unguiculus** [comment]
Marking this as stale. Please update within one week.

File: stable/aerospike/Chart.yaml (hunk 1 above)
Comment (ghost) [reply stable/aerospike/Chart.yaml:11]: thanks@unguiculus. added maintainers. apiVersion already there.

File: stable/aerospike/templates/statefulset.yaml (hunk 6 above)
Comment (ghost) [reply stable/aerospike/templates/statefulset.yaml]: @unguiculus thanks  and understandable. Just one point. Aerospike needs a set of volumes normally,.Each used for a separate namespace. If I want to follow the suggestion I still need to make it an array and use that array for defining mountVolumes and volumeClaimTemplate.

https://github.com/travelaudience/charts/blob/fce9ed49de6046c9916cdc49b084ded392bb1377/stable/aerospike/templates/statefulset.yaml#L41

for example:

```yaml
stafulset:

{{- if .Values.persistentVolume.enabled }}
  volumeClaimTemplates:
  {{- range $pv := .Values.persistentVolume.volumes }}
  - metadata:
      name: {{ $pv.name | quote }}
      annotations:
      {{- range $key, $value := $pv.annotations }}
        {{ $key }}: {{ $value }}
      {{- end }}
      spec:
        accessModes:
        {{- range $pv.accessModes }}
          - {{ . | quote }}
        {{- end }}
        resources:
          requests:
            storage: {{ $pv.size | quote }}
        {{- if $pv.storageClass }}
        {{- if (eq "-" $pv.storageClass) }}
        storageClassName: ""
        {{- else }}
        storageClassName: "{{ $pv.storageClass }}"
        {{- end }}
        {{- end }}
        selector:
{{ toYaml $pv.selector | indent 8 }}
  {{- end }}
{{- else }}
        - name: datadir
          emptyDir: {}
{{- end }}


values: 
 
persistentVolume:
  enabled: true
  volumes:
    - mountPath: /opt/aerospike/namespace-a-data
      name: aerospike-namespace-a-data
      storageClass: -
      accessModes: [ "ReadWriteOnce" ]
      size: "500G"
      selector:
        matchLabels:
          diskname: "aerospike-namespace-a-data"
    - mountPath: /opt/aerospike/namespace-b-data
      name: aerospike-namespace-b-data
      storageClass: -
      accessModes: [ "ReadWriteOnce" ]
      size: "1500G"
      selector:
        matchLabels:
          diskname: "aerospike-namespace-b-data"
```

File: stable/aerospike/Chart.yaml (hunk 1 above)
Comment (unguiculus) [reply stable/aerospike/Chart.yaml:11]: I meant `appVersion` not `apiVersion`

File: stable/aerospike/Chart.yaml [hunk 9]
```diff
@@ -9,3 +9,6 @@ version: 0.1.5
 icon: https://s3-us-west-1.amazonaws.com/aerospike-fd/wp-content/uploads/2016/06/Aerospike_square_logo.png
 sources:
 - https://github.com/aerospike/aerospike-server
+maintainers:
+- name: Kaveh Mousavi Zamani
```
Comment (unguiculus) [inline stable/aerospike/Chart.yaml]: Use Github username

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Review by unguiculus — APPROVED** [review]

**Pull request merged**